crypto/init.c

```diff
@@ -501,7 +513,9 @@ void OPENSSL_cleanup(void)
         err_free_strings_int();
     }
 
-    CRYPTO_THREAD_cleanup_local(&threadstopkey);
+    key = destructor_key;
+    destructor_key = (CRYPTO_THREAD_LOCAL)-1;
```

Comment:
Why does it get set to -1?

Reply:
To denote the fact that it's no longer valid. Since libcrypto leaks descriptor to itself (not fan of it by the way), it won't go away from memory, so that Windows will keep invoking DLL_THREAD_DETACH and OPENSSL_thread_stop even after OPENSSL_cleanup. So that destructor_key has to go back to "impossible" value.

Reply:
I've mentioned post-cleanup case in commentary to destructor_key.